arches/app/media/js/views/components/workflows/component-based-step.js

```diff
@@ -547,13 +547,15 @@ define([
     };
 
 
-    function WorkflowComponentAbstract(componentData, previouslyPersistedComponentData, externalStepData, resourceId, title, complete, saving) {
+    function WorkflowComponentAbstract(componentData, previouslyPersistedComponentData, externalStepData, resourceId, title, complete, saving, locked, setLock) {
         var self = this;
 
         this.saving = saving;
         this.complete = complete;
         this.resourceId = resourceId;
         this.componentData = componentData;
+        this.locked = locked;
+        this.setLock = setLock;
```

Comment:
Instead of passing down `setLock` here, can this be refactored into an observable, something like `shouldLockExternalStep`. that way in the viewModel (~L592) we can subscribe to the observable to manipulate params.